wordpress/wp-content/plugins/cds-base/classes/Modules/Forms/Subscribe/Setup.php

```diff
@@ -106,27 +94,43 @@ protected function subscribe(string $email, string $listId): array
         }
     }
 
-    public function confirmSubscription(): string
+    public function confirmSubscription(): array
     {
-        if (!isset($_POST['list_manager'])) {
-            return json_encode(["error" => __("400 Bad Request", "cds-snc")]);
+
+        if (!isset($_POST['cds-form-nonce'])) {
+            $message = __('400 Bad Request', 'cds-snc');
+            return ['error' => true, "error_message" => $message];
         }
 
-        if (!wp_verify_nonce($_POST['list_manager'], 'list_manager_nonce_action')) {
-            return json_encode(["error" => __("401 Unauthorized", "cds-snc")]);
+        if (!wp_verify_nonce($_POST['cds-form-nonce'], 'cds_form_nonce_action')) {
+            $message = __('400 Bad Request', 'cds-snc');
+            return ['error' => true , "error_message" => $message];
         }
 
         if (!isset($_POST["email"]) || $_POST["email"] === "") {
-            return json_encode(["error" => __("Please complete the required field to continue", "cds-snc")]);
+            return [
+                'error' =>  true,
+                'error_message' => __(
+                    'Please complete the required field(s) to continue',
+                    'cds-snc',
+                ),
+                'keys' => ['email']
+            ];
         }
 
-        if (!isset($_POST["list_id"])) {
-            return json_encode(["error" => __("Unknown subscription", "cds-snc")]);
+        if (!isset($_POST["list_id"]) || $_POST["list_id"] === "") {
+            return [
+                'error' =>  true,
+                'error_message' => __('Unknown subscription', 'cds-snc'),
+                'keys' => ['list_id']
+            ];
         }
 
         $email = sanitize_email($_POST["email"]);
         $listId = sanitize_text_field($_POST['list_id']);
 
+        // @TODO mail people when they subscribe
```

Comment:
This should already happen via ListManager.

Reply:
Okay, good to know.